modules/distributor/config.go

```diff
@@ -64,6 +68,7 @@ func (cfg *Config) RegisterFlagsAndApplyDefaults(prefix string, f *flag.FlagSet)
 	cfg.DistributorRing.KVStore.Store = "memberlist"
 	cfg.DistributorRing.HeartbeatTimeout = 5 * time.Minute
 
+	cfg.RetryAfterOnResourceExhausted = time.Second
```

Comment:
Do you think we should default to 0?  I guess I just have concerns/questions about how agents/collectors will behave and things going awry when the new Tempo is installed.  Since tempo always returns "retryable" on rate limiting, do they have a max number of retries or will it turn into an infinite loop?

Reply:
I was hoping to default to non-zero b/c it seemed to me that the default behavior should be retrying. I do think there's some sense in defaulting to 0 to keep current behavior though which people have likely become accustomed to.

> Since tempo always returns "retryable" on rate limiting, do they have a max number of retries or will it turn into an infinite loop?

This change will make the agent/collector honor it's own [retry settings](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md) which includes a max cutoff. In practice you will see some log messages like "retrying due to resource exhausted" and then "max retries exceeded. data dropped".

Reply:
As per discussions. Changing this to default to 0 which means retries are disabled by default.